### DISCUSSION TITLE
How to reset undo log?

### DISCUSSION THREAD

**Discussion by rplantiko** [josdejong/svelte-jsoneditor#385] [Q&A] [answered]
Hi, 

I use the editor to edit different JSON files, each with a different JSON schema. The user can switch between the different JSON files with a select box. If the value changes, I update the content with the `.set( content )` method, and  update the schema validator instance with `.updateProps( { validator: newValidator } )`.  

Unfortunately, if the user chooses "Undo" after the switch of the file, the former schema validator will not be restored, resulting in a ton of error messages, since the wrong validator (the new one) will be applied to the current file (which, through the 'undo', now is the former one). 

AFAIS, the best solution would be to reset the undo log when the user chooses to a new JSON file. 

Is there an API function to achieve this?     

Thanks for any suggestion
Rüdiger

**Comment by josdejong** [comment]
You can create a new editor instance, would that work?

**Reply by rplantiko** [reply]
Hi Jos, I thought about that, but hesitated: isn't it the case that then the complete editor area would be redrawn (and flicker)? And what about performance? Memory consumption? So I read your answer as "no, there is no other way and no API function to achieve what you want (to delete the current change history)". Thanks for clearing this up.

**Reply by josdejong** [reply, marked as answer]
You don't have to worry about performance and memory unless you're recreating the editor thousands of times. If you want to be sure, you can do a small experiment to see if you can notice any impact when recreating the editor.

I don't expect that the screen will flicker when recreating the editor, though you will lose for example the scroll position that the user had. Again, you can probably validate this in a few minutes by just trying it out.

If there is enough need for control over the history, we can think through an API for that.
